lib/dynflow/executors/parallel/pool.rb

```diff
@@ -4,19 +4,16 @@ class Parallel < Abstract
       class Pool < Actor
         class JobStorage
           def initialize
-            @round_robin = RoundRobin.new
-            @jobs        = Hash.new { |h, k| h[k] = [] }
+            @jobs = []
           end
 
           def add(work)
-            @round_robin.add work.execution_plan_id unless tracked?(work)
-            @jobs[work.execution_plan_id] << work
+            @jobs << work
           end
 
           def pop
             return nil if empty?
-            execution_plan_id = @round_robin.next
-            @jobs[execution_plan_id].shift.tap { delete execution_plan_id if @jobs[execution_plan_id].empty? }
+            @jobs.shift
```

Comment:
The `return nil if empty?` is functionally equivalent to `[].shift`. Some quick testing also shows that just shifting is also a tiny bit faster in this case.

Reply:
Updated